src/mfr_patcher/__main__.py

```diff
@@ -50,6 +51,16 @@
     item_patcher = ItemPatcher(rom, loc_settings)
     item_patcher.write_items()
 
+    # get hints
+    hints = None
+    if "Hints" in patch_data:
+        hints = Hints.from_json(patch_data["Hints"])
```

Comment:
You can shorten this to `hints = Hints.from_json(patch_data.get("Hints", None)`
The second get parameter is the default variable for when the key does not exist

That way you could also inline it with the check below. Up to you if you wanna do that

Reply:
The `from_json` method doesn't accept `None` as a valid input. I'm not entirely sure about doing it this way either, but it's consistent with how the lines above handle things

Reply:
Ah ok